comp/process/profiler/profiler.go

```diff
@@ -48,11 +48,8 @@ func newProfiler(deps dependencies) Component {
 			// starting the process-agent.
 			return nil
 		},
-		OnStop: func(context.Context) error {
-			profiling.Stop()
-			return nil
-		},
-	})
```

Comment:
I don't have much context on FX but aren't you changing the behaviour here by passing `profiling.Stop` instead of wrapping it around a function that always returns `nil`?

Reply:
We do indeed, good catch. Were there any reason why we didn't want to raise the error to FX ?

Reply:
Looking at the `StartStopHook` function in fx, the library is already doing the wrapper with something like:
```golang
func(context.Context) error {
    fn()
    return nil
}
```

https://github.com/uber-go/fx/blob/master/lifecycle.go#L122-L126
https://github.com/uber-go/fx/blob/master/internal/lifecycle/lifecycle.go#L78-L82

So we're actually keeping the same behaviour with the `StartStopHook` call.